src/sentry/integrations/slack/views/link_identity.py

```diff
@@ -74,6 +75,10 @@ def handle(self, request: Request, signed_params: str) -> Response:
             Identity.objects.reattach(idp, params["slack_id"], request.user, defaults)
 
         send_slack_response(integration, SUCCESS_LINKED_MESSAGE, params, command="link")
+        if not NotificationSetting.objects.has_any_provider_settings(
```

Comment:
I guess this is happening right after they link their identity, but in what case would a user have Slack settings at this point? If they had unlinked and are re-linking?

Reply:
Yeah if the user has settings and then unlinks and relinks they won't get the notifications. Also if a user links a second identity somehow they could have notification settings.